src/graphql/data-sources/serlo.ts

```diff
@@ -522,7 +245,7 @@ export class SerloDataSource extends RESTDataSource {
         },
       }
     )) as unknown
-    return await (this[setter] as (data: unknown) => Promise<T>)(data)
+    return (this.setCache(cacheKey, data as string) as unknown) as Promise<T>
```

Comment:
Almost right, seems like the only issue is the typing (which also seems to be broken on master). `data` isn't a string in all cases. So we need to change the signature from `setCache` to:
```ts
  public async setCache<T>(key: string, value: T) {
    await this.environment.cache.set(key, value)
    return value
  }
```
Then you should be able to remove the type assertions here:
```suggestion
    return this.setCache(cacheKey, data as string)
```